FogCreek/Glitch-Community
# Neighborly pull: Fetch missing permissions for search page

## Links
* https://neighborly-pull.glitch.me

## GIF/Screenshots:
![image](https://user-images.githubusercontent.com/6620164/61001295-5bc4f100-a32d-11e9-8cb3-04c2cdf3f298.png)

## Changes:
* Previously for project items on the search page we were passing the members array, but that array did not include any information regarding admin vs not admin status so our project options popover was giving incorrect options to the user. For example users were permitted to leave projects when they were the last admin on a project, (which is not something we want to encourage as we don't want orphaned projects). 
* Now if we try to render a project item with an empty permissions array, we'll try to fetch information on that project to get the permissions. 

## How To Test:
* search for a project that you are the last admin on, make sure you can't leave the project. 

## Feedback I'm looking for:
* arguably you should be able to delete the project from this page, but that's a current issue in prod, could be solved in another pr
* I'm a little worried we're making too many network calls. Perhaps the data calls for this should be at a higher level and then pulled in with something like ProjectContext ? I was mostly following the format we have with `<ProfileListLoader>` but maybe now is the time to rethink this? Feel free to tell me your thoughts. Arguably this is more of a backend issue? Perhaps I should ask our api team to change the project details we get back from algolia? I wasn't sure if that was some kind of limitation.
